### PR TITLE
feat: [CDS-78181]: support of artifactFilter in artifactory artifact

### PR DESCRIPTION
**Pull request by satyam-harness**
### Summary
tickets - https://harness.atlassian.net/browse/CDS-78181
https://harness.atlassian.net/browse/CDS-78182

Requirement - To fetch artifact lists earlier we use to provide artifact directory, but it returns only those artifacts which are available in that directory, any artifact available inside any subdirectory would not be fetched, so we wanted to add another field of artifactFilter which will basically take a regex, and any artifact matching that regex will be returned.
#### Screenshots

https://github.com/harness/harness-core-ui/assets/96036463/5d417703-8705-4d9a-8fe8-2bb776adbdfa

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts [hunk 1]
```diff
@@ -325,6 +325,17 @@ export const tagOptions: IOptionProps[] = [
   }
 ]
 
+export const filterTypeOptions: IOptionProps[] = [
+  {
+    label: 'Artifact Directory',
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts:330]: These label values should be added to YAML.

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts [hunk 2]
```diff
@@ -72,6 +72,11 @@ export enum PackageSourceTypes {
   User = 'user'
 }
 
+export enum FILTER_TYPES {
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts:75]:
```suggestion
export enum ARTIFACT_FILTER_TYPES {
```

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts [hunk 3]
```diff
@@ -158,12 +159,20 @@ export const helperTextData = (
             packageName: formik.values?.spec?.packageName
           }
     case ENABLED_ARTIFACT_TYPES.ArtifactoryRegistry:
-      return {
-        artifactPath: formik.values?.artifactPath,
-        repository: formik.values?.repository,
-        connectorRef: connectorIdValue,
-        artifactDirectory: formik.values?.artifactDirectory
-      }
+      return formik.values?.filterType === FILTER_TYPES.FILTER
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts:162]: Common out the overlapping fields here

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts (hunk 1 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts:330]: complete file `src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts` doest use strings, we can take this as a separate task and work for whole file

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts (hunk 3 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts:167]: addressed

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts (hunk 2 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts]: okay

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts (hunk 1 above)
Comment (pankajsingh-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts:330]: Please create a ticket for the same

File: src/modules/70-pipeline/utils/__tests__/stageHelpers.test.ts [hunk 4]
```diff
@@ -371,14 +371,6 @@ test('getHelpeTextForTags', () => {
     )
   ).toBe('pipeline.imagePathLabel  is  pipeline.tagDependencyRequired')
 
-  expect(
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/utils/__tests__/stageHelpers.test.ts:374]: Why these validation test removed?

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/70-pipeline/utils/__tests__/stageHelpers.test.ts (hunk 4 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/utils/__tests__/stageHelpers.test.ts:374]: because UI is now not solely dependent on artifactDirectory

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Review by Eric761 — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Pull request merged**